blocks/api/registration.js

```diff
@@ -118,15 +118,20 @@ export function registerBlockType( name, settings ) {
 	if ( ! settings.icon ) {
 		settings.icon = 'block-default';
 	}
-	settings = {
-		name,
-		attributes: get( window._wpBlocksAttributes, name ),
+
+	const attributes = settings.attributes ?
+		settings.attributes :
+		get( window._wpBlocksAttributes, name, {} );
+
+	let block = blocks[ name ] = {
 		...settings,
+		name,
+		attributes,
 	};
 
-	settings = applyFilters( 'registerBlockType', settings, name );
+	block = applyFilters( 'registerBlockType', block, name );
 
-	return blocks[ name ] = settings;
+	return blocks[ name ] = block;
```

Comment:
This assignment seems unnecessary given `let block = blocks[ name ] = {` above.

Reply:
Good catch